tests/e2e/test_e2e_skip_trained.py

```diff
@@ -0,0 +1,244 @@
+            require_pred_for_eval=True if task_name in ["task1"] else False,
+        )
+        for task_name in ["task1"]
+    ]
+    # Build model
+
+    mtl_model = EmmentalModel(name="all", tasks=tasks)
+
+    # Create learner
+    emmental_learner = EmmentalLearner()
+
+    config = {
+        "meta_config": {"seed": 0, "verbose": False},
+        "learner_config": {
+            "n_steps": 100,
+            "epochs_learned": 0,
+            "steps_learned": 90,
+            "skip_learned_data": True,
+            "online_eval": True,
+            "optimizer_config": {"lr": 0.01, "grad_clip": 100},
+        },
+        "logging_config": {
+            "counter_unit": "epoch",
+            "evaluation_freq": 0.2,
+            "writer_config": {"writer": "json", "verbose": True},
+            "checkpointing": True,
+            "checkpointer_config": {
+                "checkpoint_path": None,
+                "checkpoint_freq": 1,
+                "checkpoint_metric": {"model/all/train/loss": "min"},
+                "checkpoint_task_metrics": None,
+                "checkpoint_runway": 1,
+                "checkpoint_all": False,
+                "clear_intermediate_checkpoints": True,
+                "clear_all_checkpoints": True,
+            },
+        },
+    }
+    Meta.update_config(config)
+
+    # Learning
+    emmental_learner.learn(
+        mtl_model,
+        [train_dataloader, dev_dataloader],
+    )
+
+    test_score = mtl_model.score(test_dataloader)
+
+    assert test_score["task1/synthetic/test/loss"] > 0.3
+
+    Meta.reset()
+    init(dirpath, use_exact_log_path=use_exact_log_path)
+
+    config = {
+        "meta_config": {"seed": 0, "verbose": False},
+        "learner_config": {
+            "n_steps": 200,
+            "epochs_learned": 0,
+            "steps_learned": 90,
+            "skip_learned_data": False,
+            "online_eval": True,
+            "optimizer_config": {"lr": 0.01, "grad_clip": 100},
+        },
+        "logging_config": {
+            "counter_unit": "epoch",
+            "evaluation_freq": 0.2,
+            "writer_config": {"writer": "tensorboard", "verbose": True},
+            "checkpointing": True,
+            "checkpointer_config": {
+                "checkpoint_path": None,
+                "checkpoint_freq": 1,
+                "checkpoint_metric": {"model/all/train/loss": "min"},
+                "checkpoint_task_metrics": None,
+                "checkpoint_runway": 1,
+                "checkpoint_all": False,
+                "clear_intermediate_checkpoints": True,
+                "clear_all_checkpoints": True,
+            },
+        },
+    }
+    Meta.update_config(config)
+
+    # Learning
+    emmental_learner.learn(
+        mtl_model,
+        [train_dataloader, dev_dataloader],
+    )
+
+    test_score = mtl_model.score(test_dataloader)
+
+    assert test_score["task1/synthetic/test/loss"] <= 0.1
```

Comment:
The idea being that as it's seen the entire data for 200 steps, it's learned something (hence lower loss) rather than before where it'll only see 10 steps? Also, the test would be better if the "n_steps" was the same. The lower loss for this second model could be because n_steps is 200 rather than 100.

Reply:
Good point! Update the test to make it clear.